src/components/Link.utils.tsx

```diff
@@ -0,0 +1,36 @@
+import { Icon } from 'design-system-react';
+import type { ReactElement } from 'react';
+
+/**
+ * Programmatically determine if a link is external to the CFPB sphere of websites
+ */
+export const isExternalLink = (targetUrl: string): boolean => {
+  let parsed;
+
+  try {
+    parsed = new URL(targetUrl);
+  } catch {
+    return false; // Internal targets will fail parsing (ex. '/home')
+  }
+
+  const externalProtocols = ['http', 'tel:', 'sms:', 'ftp:'];
+  if (externalProtocols.includes(parsed.protocol)) return true;
+
+  // Any subdomain of consumerfinance.gov or the current host
+  const isInternal = new RegExp(
```

Comment:
Heyo, let's rename this thing to something other than `isInternal` since it's a regex and not a boolean?